FlowCal/transform.py

```diff
@@ -111,9 +125,68 @@ def exponentiate(data, channels = None):
         NxD transformed flow cytometry data.
 
     """
-    # Transform and return
-    def transform_fxn(x): return 10**(x/256.0)
-    return transform(data, channels, transform_fxn)
+    # Default: all channels
+    if channels is None:
+        channels = range(data.shape[1])
+
+    # Convert channels, max_range and amplification_type to iterables
+    if not hasattr(channels, '__iter__'):
+        channels = [channels]
+        if max_range is not None:
+            max_range = [max_range]
+        if amplification_type is not None:
+            amplification_type = [amplification_type]
+
+    # Convert channels to integers
+    if hasattr(data, '_name_to_index'):
+        channels = data._name_to_index(channels)
+    else:
+        channels = channels
+
+    # If max_range is not specified, take it from data.domain
+    if max_range is None:
+        if hasattr(data, 'domain'):
+            max_range = [len(data.domain(channel)) for channel in channels]
+        else:
+            raise ValueError('max_range should be specified')
+    
+    # If amplification_type is not specified, take it from
+    # data.amplification_type
+    if amplification_type is None:
+        if hasattr(data, 'domain'):
+            amplification_type = data.amplification_type(channels)
+        else:
+            raise ValueError('amplification_type should be specified')
```

Comment:
you check for `domain` but then don't use it. looks like you should be checking something else.